Internationalize strings

Wooder opened the pull request:
Hi @JamesSedlacek!

This PR adds localization to the OnboardingKit library:

- Localized strings in BottomSection and TitleSection
- Added Localizable.strings files for English (en) and German (de)

Looking forward to your feedback!

JamesSedlacek commented:
This is awesome! Thanks for taking the time to work on this.
Are you able to use string catalogs instead?
Can you update the README, so other people will know about this?

Wooder commented:
Thank you for the feedback! 🙌

I’ve implemented the requested changes:

- Switched to using string catalogs for improved i18n management
- Updated the README to include information about the currently supported languages and i18n
